src/main/java/viscount/gui/MainWindow.java

```diff
@@ -0,0 +1,81 @@
+package viscount.gui;
+
+import java.io.IOException;
+
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+import viscount.Main;
+import viscount.Viscount;
+
+/**
+ * Represents the main window component of the GUI.
+ *
+ * Handles user input and display of Viscount's responses on the GUI.
+ */
+public class MainWindow extends AnchorPane {
+    private static final String DATA_DIRECTORY_PATH = System.getProperty("user.dir") + "/data/";
+
+    @FXML
+    private ScrollPane scrollPane;
+    @FXML
+    private VBox dialogContainer;
+    @FXML
+    private TextField userInput;
+    @FXML
+    private Button sendButton;
+
+    private Viscount viscount;
+
+    /**
+     * Instantiates a new MainWindow component.
+     */
+    public MainWindow() {
+        try {
+            this.viscount = new Viscount(DATA_DIRECTORY_PATH);
+            FXMLLoader fxmlLoader = new FXMLLoader(Main.class.getResource("/view/MainWindow.fxml"));
+            fxmlLoader.setController(this);
+            fxmlLoader.setRoot(this);
+            fxmlLoader.load();
```

Comment:
I realized this part is quite different from the example. Could you briefly explain why you created a new fxmlLoader object inside the MainWindow class instead of the Main class?